### PR TITLE
RavenDB-8282 Throwing a dedicated exception if a newly created write …

### PR DESCRIPTION
**Pull request by arekpalinski**
…tx was not able to acquire the read access of the flushing in progress lock

### CODE REVIEW

File: src/Voron/StorageEnvironment.cs [hunk 1]
```diff
@@ -525,15 +525,23 @@ internal LowLevelTransaction NewLowLevelTransaction(TransactionPersistentContext
                     var wait = timeout ?? (Debugger.IsAttached ? TimeSpan.FromMinutes(30) : TimeSpan.FromSeconds(30));
 
                     if (FlushInProgressLock.IsWriteLockHeld == false)
+                    {
                         flushInProgressReadLockTaken = FlushInProgressLock.TryEnterReadLock(wait);
+                        if (flushInProgressReadLockTaken == false)
+                        {
+                            GlobalFlushingBehavior.GlobalFlusher.Value.MaybeFlushEnvironment(this);
```
Comment (arekpalinski) [inline src/Voron/StorageEnvironment.cs:532]: I've just split the previous condition into two separate once in order to be able to throw a better error here. However I _think_ we shouldn't call `MaybeFlushEnvironment` here since the flushing is already in progress.

@ayende ?

File: src/Voron/StorageEnvironment.cs [hunk 2]
```diff
@@ -480,7 +480,7 @@ private void ThrowInvalidDisposeDuringActiveTransactions(List<ActiveTransaction>
                 $"Could not dispose the environment {Options.BasePath} after {Options.DisposeWaitTime} because there are running transaction.{Environment.NewLine}" +
                 $"Either you have long running transactions or hung transactions. Can\'t dispose the environment because that would invalid memory regions{Environment.NewLine}" +
                 $"that those transactions are still looking at.{Environment.NewLine}" +
-                $"There are {activeTxs.Count:#,#0} transactions ({string.Join(", ", activeTxs)})"
+                $"There are {activeTxs.Count:#,#} transactions ({string.Join(", ", activeTxs)})"
```
Comment (ayende) [inline src/Voron/StorageEnvironment.cs:483]: Why did you remove the formatting here?

File: src/Voron/StorageEnvironment.cs (hunk 1 above)
Comment (ayende) [reply src/Voron/StorageEnvironment.cs:532]: The reason we are calling it is because we might have start flushing, had some tx in the meantime and now want to flush again, but didn't have a chance to, so this will cause us to effectively loop until we get it.

File: src/Voron/StorageEnvironment.cs (hunk 2 above)
Comment (arekpalinski) [reply src/Voron/StorageEnvironment.cs:483]: Mistake. Will fix it in a moment.

File: src/Voron/StorageEnvironment.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Voron/StorageEnvironment.cs:532]: OK
